File: coding/20th August 2025/Valid Anagram.java
//Valid Anagram
class Solution 
{
    public boolean isAnagram(String s, String t) 
    {
        if(s.length() != t.length()) return false;
        HashMap<Character, Integer> map = new HashMap<Character, Integer>();
        for(char ch : s.toCharArray())
        {
            if(map.containsKey(ch))
                map.put(ch, map.get(ch) + 1);
            elsea
                map.put(ch, 1);
        }
        for(char ch : t.toCharArray())
        {
            if(!map.containsKey(ch) || map.get(ch) <= 0)
                return false;
            map.put(ch, map.get(ch) - 1);
        }
        return true;
    }
}

File: Coding in Python/26th December 2025/Valid Anagram.py
#Valid Anagram
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        '''
        s = "".join(sorted(s))
        t = "".join(sorted(t))
        return s == t
        '''
        if len(s) != len(t):  
            return False
        arr = [0] * 26
        for i in range(len(s)):
            arr[ord(s[i]) - 97] += 1
            arr[ord(t[i]) - 97] -= 1
        for i in range(26):
            if arr[i] != 0:
                return False
        return True
